### PR TITLE
[14.0] account_asset_management: add reporting

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Port from https://github.com/OCA/account-financial-tools/pull/1123. Note that the report class had to be refactored somewhat because it is no longer allowed to patch temporary data onto `self`. I have kept this refactoring in the second commit, together with the additional tests.

### CODE REVIEW

File: account_asset_management/report/account_asset_report_xls.py [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+            and asset.date_remove >= wiz.date_from
+            and asset.date_remove <= wiz.date_to
+        )
+
+    def _group_assets(self, assets, group, grouped_assets):
+        if group:
+            group_assets = assets.filtered(lambda r: group in r.group_ids)
+        else:
+            group_assets = assets
+        group_assets = group_assets.sorted(lambda r: (r.date_start or "", r.code))
```
Comment (JordiMForgeFlow) [inline account_asset_management/report/account_asset_report_xls.py]:
```suggestion
        group_assets = group_assets.sorted(lambda r: (r.date_start or "", r.code or "", r.name))
```
#1156

Comment (StefanRijnhart) [reply account_asset_management/report/account_asset_report_xls.py]: Thanks! I've cherry picked the commit from the 13.0 branch.

**Comment by StefanRijnhart** [comment]
@JordiMForgeFlow I see you have been very active improving this module in 13.0. Did you come across this PR's new Travis error by any chance?

```ERROR: TestAssetManagement.test_10_asset_from_invoice_product_item
Traceback (most recent call last):
  File "/home/travis/build/OCA/account-financial-tools/setup/account_asset_management/odoo/addons/account_asset_management/tests/test_account_asset_management.py", line 497, in test_10_asset_from_invoice_product_item
    invoice.action_post()
  File "/home/travis/build/OCA/account-financial-tools/setup/account_asset_management/odoo/addons/account_asset_management/models/account_move.py", line 64, in action_post
    super().action_post()
  File "/home/travis/OCB-14.0/addons/account/models/account_move.py", line 2540, in action_post
    self._post(soft=False)
  File "/home/travis/OCB-14.0/addons/account_edi/models/account_move.py", line 135, in _post
    posted = super()._post(soft=soft)
  File "/home/travis/OCB-14.0/addons/account/models/account_move.py", line 2462, in _post
    raise UserError(_("The Bill/Refund date is required to validate this document."))
odoo.exceptions.UserError: The Bill/Refund date is required to validate this document.
```

**Comment by JordiMForgeFlow** [comment]
It seems it has to do with this commit https://github.com/OCA/OCB/commit/fcaa54939e9a4f0dd5e47cd0ccffe7aa24bd451c

It has not been done in 13.0 so I got no error.

**Comment by StefanRijnhart** [comment]
@JordiMForgeFlow Thanks! I've created https://github.com/OCA/account-financial-tools/pull/1159 for that.

**Comment by JordiMForgeFlow** [comment]
@StefanRijnhart Awesome! Thank you! :)

**Review by JordiMForgeFlow — APPROVED** [review]
LGTM 👍🏼

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by MiquelRForgeFlow** [comment]
Could we merge this?

**Comment by StefanRijnhart** [comment]
Yes, alright

/ocabot merge minor

**Comment by bosd** [comment]
These 2 new introduced dependencies are a tough one! Took me some troubleshooting to see why account recconcile widget broke. Seems this commit is the cause. As the module did'nt get updated correctly due to the newly introduced dependencies.
Solved now by adding the xlsx dependencies.  

(Wish there was an easier way to find out a missing dependencie instead of a broken system :wink: )

**Comment by StefanRijnhart** [comment]
Yes, sorry about that. For cases like this (and other cases as well to be honest) it is to be recommended that at least on critical installations to fix the exact module versions (cq. branch commits) that are deployed, and then scan any changes when updating the versions.

**Pull request merged**